app/models/queries/work_packages/filter/project_filter.rb

```diff
@@ -59,13 +59,35 @@ def value_objects
     available_projects = visible_projects.index_by(&:id)
 
     values
-      .map { |project_id| available_projects[project_id.to_i] }
+      .flat_map { |project_id| expanded_subprojects(available_projects[project_id.to_i]) }
       .compact
+      .uniq
+  end
+
+  def values_replaced
+    value_objects.map(&:id)
```

Comment:
I'm not too happy with having to expand all projects here, but I couldn't find a way to get descendants from multiple projects in one go. We could merge multiple `project.descendants.visible` into one query, but I don't think there is much to gain this way.